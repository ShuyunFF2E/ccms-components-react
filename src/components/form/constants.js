import { Children } from 'react';

export const LAYOUT_TYPES = {
	HORIZONTAL: 'horizontal',
	VERTICAL: 'vertical',
	INLINE: 'inline'
};

export const LABEL_ALIGN = {
	LEFT: 'left',
	RIGHT: 'right'
};

export const DATA_FIELD = 'data-field';

export const findFieldsName = (children = []) => {
	const result = [];

	Children.toArray(children).forEach(child => {
		const { props } = child;

		if (props) {
			if (props[DATA_FIELD]) {
				result.push(props[DATA_FIELD]);
			} else if (props.children && Children.count(props.children || [])) {
				result.push(...findFieldsName(props.children));
			}
		}
	});

	return result;
};

export const getNamesByNode = parentNode => {
	const children = parentNode.querySelectorAll(`[${DATA_FIELD}]`);

	return [...children].map(child => child.getAttribute(DATA_FIELD));
};

export const findDestroyedFields = (prevNames, names) => {
	return prevNames.reduce((acc, n) => {
		if (!names.includes(n)) acc.push(n);

		return acc;
	}, []);
};
