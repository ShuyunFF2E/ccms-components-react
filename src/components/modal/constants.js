export const sandboxSelector = 'r-w_s-';

export default {};
