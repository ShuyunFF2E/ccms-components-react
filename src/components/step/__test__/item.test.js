import React from 'react';
import { render, mount } from 'enzyme';
import { prefixCls } from '@utils';
import Step from '../index';
import { WAIT, FINISH, PROCESS, VERTICAL } from '../constants';

const classSelector = `${prefixCls}-step`;
const STATUS = [WAIT, FINISH, PROCESS];
const { Item } = Step;

describe('StepItem', () => {
	it('renders different status correctly', () => {
		STATUS.forEach(status => {
			const wrapper = render(
				<Step>
					<Item title="把冰箱门打开" status={status} />
				</Step>
			);
			expect(wrapper).toMatchSnapshot();
		});
	});

	it('title work correctly', () => {
		const wrapper = mount(
			<Step>
				<Item title="把冰箱门打开" />
			</Step>
		);
		expect(
			wrapper
				.find(`.${classSelector}-title`)
				.at(0)
				.text()
		).toEqual('把冰箱门打开');
	});

	it('content work correctly', () => {
		const wrapper = mount(
			<Step direction={VERTICAL}>
				<Item title="把冰箱门打开" content="步骤一" />
			</Step>
		);
		expect(
			wrapper
				.find(`.${classSelector}-content`)
				.at(0)
				.text()
		).toEqual('步骤一');
	});

	it('should content not exist when direction horizontal', () => {
		const wrapper = mount(
			<Step>
				<Item title="把冰箱门打开" content="步骤一" />
			</Step>
		);
		expect(wrapper.find(`.${classSelector}-content`)).toHaveLength(0);
	});

	it('className work correctly', () => {
		const wrapper = mount(
			<Step>
				<Item title="把冰箱门打开" />
				<Item title="把大象放进去" className="step-item" />
			</Step>
		);
		expect(
			wrapper
				.find(`.${classSelector}-item`)
				.at(1)
				.hasClass('step-item')
		).toBeTruthy();
	});

	it('onClick work correctly', () => {
		const onClick = jest.fn();
		const wrapper = mount(
			<Step>
				<Item title="把冰箱门打开" onClick={onClick} />
				<Item title="把大象放进去" />
			</Step>
		);
		wrapper
			.find(`.${classSelector}-icon`)
			.at(0)
			.simulate('click');

		expect(onClick).toHaveBeenCalledWith(0);
	});
});
