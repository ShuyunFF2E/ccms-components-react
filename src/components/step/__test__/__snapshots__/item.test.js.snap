// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepItem renders different status correctly 1`] = `
<div
  class="cloud-step horizontal circle"
  current="0"
>
  <div
    class="cloud-step-item wait"
  >
    <div
      class="cloud-step-icon wait"
    >
      <span>
        1
      </span>
    </div>
    <div
      class="cloud-step-body"
    >
      <div
        class="cloud-step-title"
      >
        把冰箱门打开
      </div>
    </div>
  </div>
</div>
`;

exports[`StepItem renders different status correctly 2`] = `
<div
  class="cloud-step horizontal circle"
  current="0"
>
  <div
    class="cloud-step-item finish"
  >
    <div
      class="cloud-step-icon finish"
    >
      <i
        class="cloud-icon cloud-icon-finish"
      />
    </div>
    <div
      class="cloud-step-body"
    >
      <div
        class="cloud-step-title"
      >
        把冰箱门打开
      </div>
    </div>
  </div>
</div>
`;

exports[`StepItem renders different status correctly 3`] = `
<div
  class="cloud-step horizontal circle"
  current="0"
>
  <div
    class="cloud-step-item process"
  >
    <div
      class="cloud-step-icon process"
    >
      <span>
        1
      </span>
    </div>
    <div
      class="cloud-step-body"
    >
      <div
        class="cloud-step-title"
      >
        把冰箱门打开
      </div>
    </div>
  </div>
</div>
`;
